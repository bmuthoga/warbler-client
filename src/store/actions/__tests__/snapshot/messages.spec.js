import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {
  loadMessages,
  remove
} from '../../messages'

describe('Messages Action Creators Snapshot Tests', () => {
  let store
  const mockStore = configureMockStore([thunk])

  beforeEach(() => {
    store = mockStore({})
  })

  it('should create LOAD_MESSAGES action if loading messages is successful', () => {
    const messages = [
      {
        _id: '8995ijg85ug8u58g8g858',
        text: 'First sample message',
        user: {
          _id: '7f99fd9r7f800g8g8g9g',
          username: 'firstTestUser',
          profileImageUrl: 'http://lorempixel.com/200/200/'
        },
        createdAt: '2019-11-23T16:45:46.174Z',
        updatedAt: '2019-11-23T16:45:46.174Z'
      },
      {
        _id: '3h5k5h45k6g6g6k6kj6h6k',
        text: 'Second sample message',
        user: {
          _id: '4vb5kl97dsfu485gjig',
          username: 'secondTestUser',
          profileImageUrl: 'http://lorempixel.com/200/200/'
        },
        createdAt: '2019-11-23T16:45:46.174Z',
        updatedAt: '2019-11-23T16:45:46.174Z'
      }
    ]
    store.dispatch(loadMessages(messages))
    expect(store.getActions()).toMatchSnapshot()
  })

  it('should create REMOVE_MESSAGE action if removing a message is successful', () => {
    const messageId = '484hjf8r99g94jg9jg'
    store.dispatch(remove(messageId))
    expect(store.getActions()).toMatchSnapshot()
  })  
})
