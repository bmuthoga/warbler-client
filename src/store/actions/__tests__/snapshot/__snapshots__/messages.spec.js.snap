// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages Action Creators Snapshot Tests should create LOAD_MESSAGES action if loading messages is successful 1`] = `
Array [
  Object {
    "messages": Array [
      Object {
        "_id": "8995ijg85ug8u58g8g858",
        "createdAt": "2019-11-23T16:45:46.174Z",
        "text": "First sample message",
        "updatedAt": "2019-11-23T16:45:46.174Z",
        "user": Object {
          "_id": "7f99fd9r7f800g8g8g9g",
          "profileImageUrl": "http://lorempixel.com/200/200/",
          "username": "firstTestUser",
        },
      },
      Object {
        "_id": "3h5k5h45k6g6g6k6kj6h6k",
        "createdAt": "2019-11-23T16:45:46.174Z",
        "text": "Second sample message",
        "updatedAt": "2019-11-23T16:45:46.174Z",
        "user": Object {
          "_id": "4vb5kl97dsfu485gjig",
          "profileImageUrl": "http://lorempixel.com/200/200/",
          "username": "secondTestUser",
        },
      },
    ],
    "type": "LOAD_MESSAGES",
  },
]
`;

exports[`Messages Action Creators Snapshot Tests should create REMOVE_MESSAGE action if backend API successfully deletes a message 1`] = `
Array [
  Object {
    "error": "connect ECONNREFUSED 127.0.0.1:80",
    "type": "ADD_ERROR",
  },
]
`;

exports[`Messages Action Creators Snapshot Tests should create REMOVE_MESSAGE action if removing a message is successful 1`] = `
Array [
  Object {
    "id": "484hjf8r99g94jg9jg",
    "type": "REMOVE_MESSAGE",
  },
]
`;
